sklearn/pipeline.py

```diff
@@ -575,18 +603,21 @@ def predict(self, X, **params):
         y_pred : ndarray
             Result of calling `predict` on the final estimator.
         """
-        Xt = X
+        with _handle_warnings(self):
```

Comment:
It might be worth to also add some `TODO` next to each context manager to have more occurences.

Reply:
Maybe having a decorator instead of a context manager would avoid the extra indentation?

Reply:
these methods are complicated, they already have decorators, and adding another decorator might complicate things. So I rather do the dirty-ish thing here and keep it as is.